ethclient/ethclient.go

```diff
@@ -265,7 +265,7 @@ func (ec *client) getBlock(ctx context.Context, method string, args ...interface
 		}
 		txs[i] = tx.tx
 	}
-	return types.NewBlockWithHeader(head).WithBody(txs, uncles).WithExtData(body.Version, (*[]byte)(body.BlockExtraData)), nil
+	return types.NewBlockWithHeader(head).WithBody(types.Body{Transactions: txs, Uncles: uncles}), nil
```

Comment:
why don't we use body here? seems we'd miss `WithExtData(body.Version, (*[]byte)(body.BlockExtraData))`

Reply:
> seems we'd miss `WithExtData(body.Version, (*[]byte)(body.BlockExtraData))`

I agree. The extra needs to be set on the newly constructed `Block`. There's no point setting it on the ephemeral `Body` being passed to `WithBody` as that will just be cloned.

Reply:
I'm not sure what the difference between calling .WithBody(body) and the code here is?
body already includes the extras, and calling WithBody also clones the extras.
Since this is shared code I would err on the side of matching upstream (as it will be in future versions)

Reply:
The `body` variable is an `rpcBlock` type 😉 (easy to miss out! - maybe a bit of a bad var name, but that's out of scope I'd say)